### DISCUSSION TITLE
Pixel Saver / Unite and similar gnome extensions problem

### DISCUSSION THREAD

**Discussion by Froggy232** [netblue30/firejail#4110] [Q&A] [answered]
Hi,
I have an archlinux + gnome shell installation, and I have installed the Unite extension to have app open in full screen, like this : https://extensions.gnome.org/extension/723/pixel-saver/

The extension work very well with no firejailed app, but it doesn't work with firejail. Would you have an idea of how to make it work? Do you know which security option I would need to disable to make it work?

I don't think any log file would be relevant, but please feel free to ask for anything I could provide.
Thanks for your work anyway, it works perfectly except that!
Best regards

**Comment by rusty-snake** [comment, marked as answer]
**Fix: `${HOME}/.local/share/gnome-shell` must be allowed.**
For that you need to `noblacklist ${HOME}/.local/share/gnome-shell` if the profile `include`s `disable-common.inc` (the most profiles) and `whitelist ${HOME}/.local/share/gnome-shell` for `withlist`ing profiles. If a profiles has both, `include disable-common.inc` and `include whitelist-common.inc`, you need to add both:

    noblacklist ${HOME}/.local/share/gnome-shell
    whitelist ${HOME}/.local/share/gnome-shell

--------

Original post:

This sound very much like D-Bus. Try `ignore dbus-user none`/`ignore dbus-user filter`.

If this works, dig deeper with `busctl --user list --no-legend --activatable --acquired | grep -i pixel` and `busctl --user tree org.gnome.Shell`. Does the later have a object of this extension or the first a name? Add  talk rules for them.

Since it has no Wayland support and does everything with X11 technologies, it could also be the userns or pidns. Try `ignore noroot` or start with `--noprofile`. If it still broken with `--noprofile`, there is no fix.

**Comment by Froggy232** [comment]
Thanks you a lot!
I have tried `--noprofile` and it worked, but I don't understand in what file I need to put the other command exactly?
Thanks again

**Reply by rusty-snake** [reply]
`--noprofile` is just a debug option to see whether there are bigger problems going on. It's no much difference to not using firejail.

> I don't understand in what file I need to put the other command exactly

FYI only: Just create a empty profile `mkdir -p ~/.config/firejail && touch ~/.config/firejail/gnome-calculator.profile`.

**Comment by Froggy232** [comment]
I tried to add `ignore dbus-user none` `ignore dbus-user filter` and `ignore noroot` to a file "liferea.local", but it didn't change anything.
Was it the good method?

**Reply by rusty-snake** [reply]
If there is nothing useful in your journal, copy the profile to you home, comment all lines and start removing the comments line for line (test command: `firejail --profile=./liferea.profile liferea.profile`).

**Reply by Froggy232** [reply]
Thanks, I will try that right now

Edit : It work if I comment everything, I will uncomment line by line and answer back

**Reply by Froggy232** [reply]
So, I tried it and there is two things which prevent it from working, at least on liferea : 
- First is the line "include disable-common.inc", which I suppose is important
- Second is this block of lines : 
```#mkdir ${HOME}/.cache/liferea
#mkdir ${HOME}/.config/liferea
#mkdir ${HOME}/.local/share/liferea
#whitelist ${HOME}/.cache/liferea
#whitelist ${HOME}/.config/liferea
#whitelist ${HOME}/.local/share/liferea
#whitelist /usr/share/liferea
#include whitelist-common.inc
#include whitelist-usr-share-common.inc
#include whitelist-var-common.inc
```
Which is very weird I think.

If I comment all theses lines, everythings works.
Any idea to continue the research?
Thanks you again

**Reply by rusty-snake** [reply]
Just guessing:
```
noblacklist ${HOME}/.local/share/gnome-shell
whitelist ${HOME}/.local/share/gnome-shell
```

OT: Use `` ``` `` for code-blocks and `` ` `` for inline code.

**Reply by Froggy232** [reply]
Thanks, it was that!
It's perfect now, thanks you A LOT!
Do you think it's safe to add it to globals.local?

**Reply by Froggy232** [reply]
The `whitelist ${HOME}/.local/share/gnome-shell` line make some app not open, probably because they don't have whitelist profile?
Do you have an idea on how to apply this option automatically?

**Reply by Froggy232** [reply]
I just added the lines manually for each app where it was needed.
Thanks you a lot, wonderful app!
